app/bundles/CoreBundle/Assets/js/1.core.js

```diff
@@ -390,51 +390,50 @@ var Mautic = {
             MauticVars.ignoreIconSpin = false;
             return;
         }
-
+    
         if (typeof target == 'object' && typeof(target.target) !== 'undefined') {
             target = target.target;
         }
-
+    
         if (mQuery(target).length) {
             var hasBtn = mQuery(target).hasClass('btn');
-            var hasIcon = mQuery(target).hasClass('fa');
+            var hasIcon = mQuery(target).attr('class') && mQuery(target).attr('class').startsWith('ri-');
             var dontspin = mQuery(target).hasClass('btn-nospin');
-
-            var i = (hasBtn && mQuery(target).find('i.fa').length) ? mQuery(target).find('i.fa') : target;
-
-            if (!dontspin && ((hasBtn && mQuery(target).find('i.fa').length) || hasIcon)) {
-                var el = (hasIcon) ? target : mQuery(target).find('i.fa').first();
+    
+            var i = (hasBtn && mQuery(target).find('i[class^="ri-"]').length) ? mQuery(target).find('i[class^="ri-"]') : target;
+    
+            if (!dontspin && ((hasBtn && mQuery(target).find('i[class^="ri-"]').length) || hasIcon)) {
+                var el = (hasIcon) ? target : mQuery(target).find('i[class^="ri-"]').first();
                 var identifierClass = (new Date).getTime();
                 MauticVars.iconClasses[identifierClass] = mQuery(el).attr('class');
-
-                var specialClasses = ['fa-fw', 'ri-lg', 'ri-2x', 'ri-3x', 'ri-4x', 'ri-5x', 'fa-li', 'text-white', 'text-muted'];
+    
+                var specialClasses = ['ri-fw', 'ri-lg', 'ri-2x', 'ri-3x', 'ri-4x', 'ri-5x', 'ri-li', 'text-white', 'text-muted'];
                 var appendClasses = "";
-
-                //check for special classes to add to spinner
+    
                 for (var i = 0; i < specialClasses.length; i++) {
                     if (mQuery(el).hasClass(specialClasses[i])) {
                         appendClasses += " " + specialClasses[i];
                     }
                 }
                 mQuery(el).removeClass();
-                mQuery(el).addClass('fa fa-spinner fa-spin ' + identifierClass + appendClasses);
+                mQuery(el).addClass('ri-loader-line ri-spin ' + identifierClass + appendClasses);
+                mQuery(el).removeClass();
+                mQuery(el).addClass('ri-loader-line ri-spin ' + identifierClass + appendClasses);
             }
-        }
-    },
```

Comment:
Is this really working for you? I'm getting syntax error
![Screenshot 2024-07-12 at 10 17 15](https://github.com/user-attachments/assets/d8626688-1b5b-43bc-b92c-b89dd5a71fe5)

The steps to test could be that throttling is set to slow 3G in dev tools and then change the date range in the dashboard to obfuscate the cache and reload the widgets.

Reply:
How can I check in a practical scenario?
Where can I trigger this loading icon?